pkg/controller/infrastructure/client.go

```diff
@@ -15,89 +15,68 @@
 package infrastructure
 
 import (
+	"context"
 	"fmt"
 	"strings"
 
 	alicloudclient "github.com/gardener/gardener-extension-provider-alicloud/pkg/alicloud/client"
-
-	"github.com/aliyun/alibaba-cloud-sdk-go/services/vpc"
 )
 
 // GetVPCInfo gets info of an existing VPC.
-func GetVPCInfo(vpcClient alicloudclient.VPC, vpcID string) (*VPCInfo, error) {
-	describeVPCsReq := vpc.CreateDescribeVpcsRequest()
-	describeVPCsReq.VpcId = vpcID
-	describeVPCsRes, err := vpcClient.DescribeVpcs(describeVPCsReq)
+func GetVPCInfo(ctx context.Context, vpcClient alicloudclient.VPC, vpcID string) (*VPCInfo, error) {
+	vpc, err := vpcClient.GetVPCWithID(ctx, vpcID)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(describeVPCsRes.Vpcs.Vpc) != 1 {
-		return nil, fmt.Errorf("ambiguous VPC response: expected 1 VPC but got %v", describeVPCsRes.Vpcs.Vpc)
+	if len(vpc) != 1 {
+		return nil, fmt.Errorf("ambiguous VPC response: expected 1 VPC but got %v", vpc)
 	}
 
-	vpcCIDR := describeVPCsRes.Vpcs.Vpc[0].CidrBlock
-
-	describeNATGatewaysReq := vpc.CreateDescribeNatGatewaysRequest()
-	describeNATGatewaysReq.VpcId = vpcID
-	describeNatGatewaysRes, err := vpcClient.DescribeNatGateways(describeNATGatewaysReq)
+	natGateway, err := vpcClient.GetNatGatewayWithVPCID(ctx, vpcID)
 	if err != nil {
 		return nil, err
 	}
 
-	if len(describeNatGatewaysRes.NatGateways.NatGateway) != 1 {
-		return nil, fmt.Errorf("ambiguous NAT Gateway response: expected 1 NAT Gateway but got %v", describeNatGatewaysRes.NatGateways.NatGateway)
+	if len(natGateway) != 1 {
+		return nil, fmt.Errorf("ambiguous NAT Gateway response: expected 1 NAT Gateway but got %v", natGateway)
 	}
 
-	natGateway := describeNatGatewaysRes.NatGateways.NatGateway[0]
-	natGatewayID := natGateway.NatGatewayId
-	sNatTableIDs := strings.Join(natGateway.SnatTableIds.SnatTableId, ",")
-
-	internetChargeType, err := FetchEIPInternetChargeType(vpcClient, vpcID)
+	internetChargeType, err := FetchEIPInternetChargeType(ctx, vpcClient, vpcID)
 	if err != nil {
 		return nil, err
 	}
 
 	return &VPCInfo{
-		CIDR:               vpcCIDR,
-		NATGatewayID:       natGatewayID,
-		SNATTableIDs:       sNatTableIDs,
+		CIDR:               vpc[0].CidrBlock,
+		NATGatewayID:       natGateway[0].NatGatewayId,
+		SNATTableIDs:       strings.Join(natGateway[0].SnatTableIds.SnatTableId, ","),
 		InternetChargeType: internetChargeType,
 	}, nil
 }
 
 // FetchEIPInternetChargeType fetches the internet charge type for the VPC's EIP.
-func FetchEIPInternetChargeType(vpcClient alicloudclient.VPC, vpcID string) (string, error) {
-	describeNATGatewaysReq := vpc.CreateDescribeNatGatewaysRequest()
-	describeNATGatewaysReq.VpcId = vpcID
-
-	describeNatGatewaysRes, err := vpcClient.DescribeNatGateways(describeNATGatewaysReq)
+func FetchEIPInternetChargeType(ctx context.Context, vpcClient alicloudclient.VPC, vpcID string) (string, error) {
+	natGateway, err := vpcClient.GetNatGatewayWithVPCID(ctx, vpcID)
```

Comment:
The interface name is tricky. I suppose it will return a `list` of natGatway. 
```suggestion
	natGateways, err := vpcClient.GetNatGatewayWithVPCID(ctx, vpcID)
```